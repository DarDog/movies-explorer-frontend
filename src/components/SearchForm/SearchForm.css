.search {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.search__form {
  width: 100%;
  display: flex;
  align-items: center;
  margin-bottom: 30px;
}

.search__input {
  width: 100%;
  margin-right: 10px;
  background-color: #F9F9F9;
  border-radius: 9px;
  border: none;
  padding: 15px 22px;
  box-sizing: border-box;
  font-weight: 400;
  font-size: 14px;
  line-height: 18px;
  color: #A0A0A0;
  outline: none;
}

.search__input:focus {
  box-shadow: 1px 1px 10px rgba(92, 92, 92, 0.25);
}

.search__submit {
  background-color: #2BE080;
  border-radius: 4px;
  border: none;
  font-weight: 500;
  font-size: 12px;
  line-height: 18px;
  text-align: center;
  color: #FFF;
  padding: 13px 20px;
  cursor: pointer;
  transition: .3s;
}

.search__submit:hover {
  transform: scale(1.1);
  opacity: .8;
}

.search__border {
  margin-top: 40px;
  width: 100%;
  border-bottom: 1px solid #E8E8E8;
}

.main__search {
  padding: 80px 70px 0;
  width: 100%;
  max-width: 1280px;
  margin-bottom: 10px;
  box-sizing: border-box;
}

.movies__not-found {
  margin: 0;
  padding: 0;
  color: #5C5C5C;
  font-size: 16px;
  line-height: 18px;
  text-align: center;
}

@media screen and (max-width: 800px){
  .main__search {
    padding: 80px 30px 0;
  }
}

@media screen and (max-width: 500px){
  .main__search {
    padding: 80px 14px 0;
  }

  .search__input {
    padding: 13px 16px;
    font-size: 12px;
    line-height: 18px;
  }

  .search__paragraph {
    font-size: 11px;
  }
}
