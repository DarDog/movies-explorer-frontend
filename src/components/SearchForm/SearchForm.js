import React, {useState} from 'react';
import './SearchForm.css'
import Switch from "../Switch/Switch";

const SearchForm = () => {
  const [movie, setMovie] = useState('');

  const handleChangeSearchInput = (evt) => {
    setMovie(evt.target.value);
  }

  return (
    <section className="search main__search">
      <form className="search__form" name='search'>
        <input type="text" className="search__input" placeholder="Фильм" required value={movie} onChange={handleChangeSearchInput}/>
        <button type="submit" className="search__submit">Найти</button>
      </form>
      <Switch/>
      <div className='search__border'/>
    </section>
  );
}

export default SearchForm;
