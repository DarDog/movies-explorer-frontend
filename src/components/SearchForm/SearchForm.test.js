import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchForm from "./SearchForm";
import '@testing-library/jest-dom';

describe('Компонент SearchForm', () => {
  it('Рендерится', () => {
    render(
      <SearchForm />
    );

    expect(screen.getByRole('form')).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByText('Найти')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(screen.getByText('Короткометражки')).toBeInTheDocument();
  })
  it('соответствует Snapshot',  () => {
    const component = render(
      <SearchForm />
    );

    expect(component).toMatchSnapshot();
  });
})
