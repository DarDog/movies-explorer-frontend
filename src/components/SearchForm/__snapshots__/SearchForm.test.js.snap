// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент SearchForm соответствует Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="search main__search"
      >
        <form
          class="search__form"
          name="search"
        >
          <input
            class="search__input"
            placeholder="Фильм"
            type="text"
          />
          <button
            class="search__submit"
            type="submit"
          >
            Найти
          </button>
        </form>
        <div
          class="search__switch-container"
        >
          <label
            class="search__switch"
          >
            <input
              class="search__checkbox"
              type="checkbox"
            />
            <span
              class="search__slider"
            />
          </label>
          <p
            class="search__paragraph"
          >
            Короткометражки
          </p>
        </div>
        <div
          class="search__border"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="search main__search"
    >
      <form
        class="search__form"
        name="search"
      >
        <input
          class="search__input"
          placeholder="Фильм"
          type="text"
        />
        <button
          class="search__submit"
          type="submit"
        >
          Найти
        </button>
      </form>
      <div
        class="search__switch-container"
      >
        <label
          class="search__switch"
        >
          <input
            class="search__checkbox"
            type="checkbox"
          />
          <span
            class="search__slider"
          />
        </label>
        <p
          class="search__paragraph"
        >
          Короткометражки
        </p>
      </div>
      <div
        class="search__border"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
